src/resonanceReconstruction/pack.hpp

```diff
@@ -1,56 +1,45 @@
-template< typename... Ts >
-struct Pack;
-
-template< typename... Ts >
-auto pack( Ts&&... ts ) -> Pack< std::decay_t<Ts>... >;
-
-template< typename... Ts >
-struct Pack {
-  std::tuple<Ts...> data;
-
-  Pack( Ts... data ) : data( std::make_tuple(data...) ){}
-  
-  static constexpr auto indices =
-    std::make_index_sequence< std::tuple_size< decltype(data) >::value >{};
-  
-  template< typename... Us, std::size_t... indices >
-  auto add( const Pack< Us... >& other, std::index_sequence< indices... > ) const {
-    static_assert( std::tuple_size< decltype( this->data ) >::value
-                   == std::tuple_size< decltype( other.data ) >::value, "" );
-    return pack( ( std::get<indices>(this->data)
-                   + std::get<indices>(other.data) )...  );
-  }
-
-  template< typename... Us, std::size_t... indices >
-  auto subtract( const Pack< Us... >& other, std::index_sequence< indices... > ) const {
-    static_assert( std::tuple_size< decltype( this->data ) >::value
-                   == std::tuple_size< decltype( other.data ) >::value, "" );
-    return pack( ( std::get<indices>(this->data)
-                              - std::get<indices>(other.data) )...  );
-  }
-
-  template< typename Scalar, std::size_t... indices >
-  auto multiply( const Scalar& scalar, std::index_sequence< indices... > ) const {
-    return pack( ( std::get<indices>(this->data) * scalar )...  );
-  }
-
-  template< typename... Us >
-  auto operator+( const Pack< Us... >& other ) const {
-    return this->add( other, indices );
-  }
-
-  template< typename... Us >
-  auto operator-( const Pack< Us... >& other ) const {
-    return this->subtract( other, indices );
-  }
-
-  template< typename Scalar >
-  auto operator*( const Scalar& scalar ) const {
-    return this->multiply( scalar, indices );
-  }
-};
+template<class... Ts>
+class Pack;
 
-template< typename... Ts >
-auto pack( Ts&&... ts )  -> Pack< std::decay_t<Ts>... > {
-  return Pack< std::decay_t<Ts>... >{ std::forward<Ts>(ts)... };
+template<class... Ts>
+inline auto pack(Ts&&... ts)
+{
+   return Pack<std::decay_t<Ts>...>{ std::forward<Ts>(ts)... };
 }
+
+template<class... Ts>
+class Pack {
```

Comment:
I can't really tell what you did differently here. The GitHub diff isn't being very helpful.

Reply:
In the case of `pack.hpp` it's really best to just do a side-by-side, as the change was largely a rewrite. It cleans things up, shortens the code a bit, and uses the new `sizeof...` to replace the previous longer way of doing it.